dali/util/make_string.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.#ifndef DALI_MAKE_STRING_H
+
+#ifndef DALI_UTIL_MAKE_STRING_H_
+#define DALI_UTIL_MAKE_STRING_H_
+
+#include <string>
+
+namespace dali {
+
+namespace detail {
+
+template <class T>
+std::string make_string(std::stringstream &ss, const T &val) {
+  ss << val;
+  return ss.str();
+}
+
+
+template <class T, class... Args>
+std::string make_string(std::stringstream &ss, const T &val, const Args &... args) {
+  ss << val;
```

Comment:
Maybe consider some kind of spacing between the arguments? Right now it might be hard to read the resulting string.

Reply:
:+1:

Reply:
I omitted the spacing intentionally, as that's string concatenation, not printing itself. I can imagine the situation, that it can be used in case, where such spacings will be no-go.

Reply:
Actually, I agree with @awolant. The common use case of this would be the one with spacing between the arguments. You can either make it configurable (passing a delimiter argument) or make a choice by default (e.g. `,` or ` `)

Reply:
Actually, I agree with @awolant here. I think the most common use case of this will be displaying a bunch of values separated by spaces. I'd rather do `make_string(str1, 2, 3)` than `make_string(str1, " ", 2, " ", 3)`. Consider making this configurable by passing a delimiter (perhaps with a default value of `" "` for the delimiter)

Reply:
That function turned out to be a `concat` ∘ `to_string`. And it will most probably be now used for most of the DALI_ENFORCE stuff to easily print messages, in that case we can do with `make_string("Images don't have matching number of channels, expected ", x, " found ", y, ".");`

Reply:
But I'am not sure if it's the best solution that we can have.

Reply:
If `vec<3>` needs to be printed then the delimiter is more than welcome. But in case of `make_string("Images don't have matching number of channels, expected ", x, " found ", y, ".");` rather not.
How about two variants:
* `make_string` without delimiter
* `make_string_delim` with delimeter
?

Reply:
Done